### PR TITLE
Fix truncate lake table bug

### PR DESCRIPTION
**Pull request by abc982627271**
## What type of PR is this：
- [x] bug
- [ ] feature
- [ ] enhancement
- [ ] refactor
- [ ] others

## Which issues of this PR fixes ：

Fixes #

## Problem Summary(Required) ：

After truncated , showing partition and tablet would return 'unknown error',  this pr fix this bug

### CODE REVIEW

**Comment by abc982627271** [comment]
run starrocks_fe_unittest

File: fe/fe-core/src/main/java/com/starrocks/catalog/PartitionInfo.java [hunk 1]
```diff
@@ -162,6 +162,14 @@ public void addPartition(long partitionId, DataProperty dataProperty,
         idToInMemory.put(partitionId, isInMemory);
     }
 
+    public void addPartition(long partitionId, DataProperty dataProperty,
+                             short replicationNum,
+                             boolean isInMemory,
+                             StorageInfo storageInfo) {
+        this.addPartition(partitionId, dataProperty, replicationNum, isInMemory);
+        idToStorageInfo.put(partitionId, storageInfo);
```
Comment (wyb) [inline fe/fe-core/src/main/java/com/starrocks/catalog/PartitionInfo.java:170]: Better to check `storageInfo` is null.

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 5 / 21 (23.81%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/load/PartitionUtils.java|0|1|00.00%|[40]|
|:large_blue_circle:|com/starrocks/catalog/ExternalOlapTable.java|0|1|00.00%|[398]|
|:large_blue_circle:|com/starrocks/server/LocalMetastore.java|1|11|09.09%|[1140, 3834, 3848, 3849, 3850, 3968, 3972, 3973, 3974, 3975]|
|:large_blue_circle:|com/starrocks/catalog/RangePartitionInfo.java|1|4|25.00%|[102, 103, 104]|
|:large_blue_circle:|com/starrocks/catalog/OlapTable.java|1|2|50.00%|[1340]|
|:large_blue_circle:|com/starrocks/catalog/PartitionInfo.java|2|2|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 5 / 22 (22.73%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/load/PartitionUtils.java|0|1|00.00%|[40]|
|:large_blue_circle:|com/starrocks/backup/RestoreJob.java|0|2|00.00%|[668, 928]|
|:large_blue_circle:|com/starrocks/catalog/ExternalOlapTable.java|0|2|00.00%|[383, 398]|
|:large_blue_circle:|com/starrocks/server/LocalMetastore.java|1|11|09.09%|[1140, 3834, 3848, 3849, 3850, 3968, 3972, 3973, 3974, 3975]|
|:large_blue_circle:|com/starrocks/catalog/OlapTable.java|1|3|33.33%|[1229, 1340]|
|:large_blue_circle:|com/starrocks/catalog/RangePartitionInfo.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/catalog/PartitionInfo.java|2|2|100.00%|[]|

File: fe/fe-core/src/main/java/com/starrocks/catalog/ExternalOlapTable.java [hunk 2]
```diff
@@ -395,7 +395,7 @@ public void updateMeta(String dbName, TTableMeta meta, List<TBackendMeta> backen
                                 singePartitionDesc.getBase_desc().getData_property().get(partitionId);
                         DataProperty dataProperty = new DataProperty(thriftDataProperty.getStorage_medium(),
                                 thriftDataProperty.getCold_time());
-                        partitionInfo.addPartition(partitionId, dataProperty, replicaNum, inMemory);
+                        partitionInfo.addPartition(partitionId, dataProperty, replicaNum, inMemory, null);
```
Comment (wyb) [inline fe/fe-core/src/main/java/com/starrocks/catalog/ExternalOlapTable.java:398]: Better to revert.

File: fe/fe-core/src/main/java/com/starrocks/server/LocalMetastore.java [hunk 3]
```diff
@@ -1137,7 +1137,7 @@ public void replayAddPartition(PartitionPersistInfo info) throws DdlException {
                         info.isInMemory());
             } else {
                 partitionInfo.addPartition(
-                        partition.getId(), info.getDataProperty(), info.getReplicationNum(), info.isInMemory());
+                        partition.getId(), info.getDataProperty(), info.getReplicationNum(), info.isInMemory(), null);
```
Comment (wyb) [inline fe/fe-core/src/main/java/com/starrocks/server/LocalMetastore.java:1140]: Better to revert.

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 7 / 19 (36.84%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/server/LocalMetastore.java|1|10|10.00%|[3834, 3848, 3849, 3850, 3968, 3972, 3973, 3974, 3975]|
|:large_blue_circle:|com/starrocks/catalog/RangePartitionInfo.java|1|3|33.33%|[108, 109]|
|:large_blue_circle:|com/starrocks/catalog/OlapTable.java|1|2|50.00%|[1339]|
|:large_blue_circle:|com/starrocks/catalog/PartitionInfo.java|4|4|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 9 / 28 (32.14%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/server/LocalMetastore.java|3|19|15.79%|[3834, 3843, 3856, 3907, 3917, 3918, 3919, 3921, 3922, 3923, 3925, 3977, 3981, 3982, 3983, 3984]|
|:large_blue_circle:|com/starrocks/catalog/RangePartitionInfo.java|1|3|33.33%|[108, 109]|
|:large_blue_circle:|com/starrocks/catalog/OlapTable.java|1|2|50.00%|[1339]|
|:large_blue_circle:|com/starrocks/catalog/PartitionInfo.java|4|4|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 7 / 27 (25.93%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/server/LocalMetastore.java|1|18|05.56%|[4107, 4116, 4175, 4176, 4177, 4188, 4189, 4190, 4192, 4193, 4194, 4196, 4248, 4252, 4253, 4254, 4255]|
|:large_blue_circle:|com/starrocks/catalog/RangePartitionInfo.java|1|3|33.33%|[110, 111]|
|:large_blue_circle:|com/starrocks/catalog/OlapTable.java|1|2|50.00%|[1332]|
|:large_blue_circle:|com/starrocks/catalog/PartitionInfo.java|4|4|100.00%|[]|

**Review by gengjun-git — APPROVED** [review]

**Review by wyb — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Comment by abc982627271** [comment]
@mergifyio backport branch-3.0

**Pull request merged**